lib/graphlient/client.rb

```diff
@@ -45,18 +47,32 @@ def http_adapter_class
     end
 
     def http(&block)
-      @http ||= http_adapter_class.new(@url, headers: @options[:headers], &block)
+      @http ||= http_adapter_class.new(@url, headers: options[:headers], &block)
     end
 
     def schema
-      @schema ||= GraphQL::Client.load_schema(http)
+      @schema ||= GraphQL::Client.load_schema(options[:schema_path]&.to_s || http)
+    end
+
+    def dump_schema
+      schema_path = options.fetch(:schema_path) { raise_missing_configuration_error }
+
+      GraphQL::Client.dump_schema(http, schema_path.to_s)
```

Comment:
You can just inline the `raise_missing_configuration_error` method:

```ruby
ERROR_MESSAGE = 'schema_cache_path is missing. Please add it like this: `Graphlient.new(url, schema_cache_path: YOUR_PATH)`'

def dump_schema
  schema_path = options[:schema_path] || raise MissingConfigurationError, ERROR_MESSAGE

  GraphQL::Client.dump_schema(http, schema_path)
end
```

Keep in mind that the `Hash#fetch` method doesn't behave exactly like the `#[]` hash access, and I think we want the former:

```ruby
{}[:foo] || raise "error"                  # => error
{ foo: nil }[:foo] || raise("error")       # => error

{}.fetch(:foo) { raise "error" }           # => error
{ foo: nil }.fetch(:foo) { raise "error" } # => nil
```